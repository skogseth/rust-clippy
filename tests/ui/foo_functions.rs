#![allow(unused)]
#![warn(clippy::foo_functions)]

// Impl methods
struct A;
impl A {
    pub fn fo(&self) {}
    pub fn foo(&self) {}
    pub fn food(&self) {}
}

// Default trait methods
trait B {
    fn fo(&self) {}
    fn foo(&self) {}
    fn food(&self) {}
}

// Plain functions
fn fo() {}
fn foo() {}
fn food() {}

fn main() {
    // We also don't want to lint method calls
    foo();
    let a = A;
    a.foo();
}
