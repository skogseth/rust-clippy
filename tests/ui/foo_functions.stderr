error: function named `foo`
  --> tests/ui/foo_functions.rs:8:5
   |
LL |     pub fn foo(&self) {}
   |     ^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider using a more meaningful name
   = note: `-D clippy::foo-functions` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::foo_functions)]`

error: function named `foo`
  --> tests/ui/foo_functions.rs:15:5
   |
LL |     fn foo(&self) {}
   |     ^^^^^^^^^^^^^^^^
   |
   = help: consider using a more meaningful name

error: function named `foo`
  --> tests/ui/foo_functions.rs:21:1
   |
LL | fn foo() {}
   | ^^^^^^^^^^^
   |
   = help: consider using a more meaningful name

error: aborting due to 3 previous errors

